runtime/compiler/env/ProcessorDetection.cpp

```diff
@@ -269,6 +269,20 @@ TR_Processor mapJ9Processor(J9ProcessorArchitecture j9processor)
       case PROCESSOR_PPC_P9:
          tp = TR_PPCp9;
          break;
+      case PROCESSOR_PPC_P10:
+         // P10 support is not yet well-tested, so it's currently gated behind an environment
+         // variable to prevent it from being used by accident by users who use old versions of
+         // OpenJ9 once P10 chips become available.
+         if (jitConfig)
+            {
+            static bool enableP10 = feGetEnv("TR_EnableExperimentalPower10Support");
+            tp = enableP10 ? TR_PPCp10 : TR_PPCp9;
```

Comment:
older version of OpenJ9 cannot recognize P10 such that it is only an Unknown processor (i.e. DefaultProcessor). This code on P10Processor never kicks in anyway such that It is unnecessary.

Reply:
Sorry, I should clarify: by "older versions", I'm referring to versions that will be released between when this PR is merged and when the P10 support is fully tested and verified. If a user were to try and use one of those builds on a newer machine with a POWER10 chip, OpenJ9 would recognize it and enable the new features, despite them having not yet undergone testing.

Reply:
providing a testing knob might make more senses, before it is enabled by default.